www/include/configuration/configObject/host_categories/DB-Func.php

```diff
@@ -112,64 +114,115 @@ function disableHostCategoriesInDB($hc_id = null, $hc_arr = array())
 {
-    global $pearDB, $centreon, $is_admin;
+    global $pearDB, $centreon;
 
     $hcAcl = array();
     foreach ($hostcategories as $key => $value) {
-        $DBRESULT = $pearDB->query("SELECT * FROM hostcategories WHERE hc_id = '".$key."' LIMIT 1");
-        $row = $DBRESULT->fetchRow();
+        $hcId = filter_var($key, FILTER_VALIDATE_INT);
+        $statement = $pearDB->prepare("SELECT * FROM hostcategories WHERE hc_id = ? LIMIT 1");
+        $result = $pearDB->execute($statement, array($hcId));
+        $row = $result->fetchRow();
         $row["hc_id"] = '';
-        for ($i = 1; $i <= $nbrDup[$key]; $i++) {
+        for ($i = 1; $i <= $nbrDup[$hcId]; $i++) {
             $val = null;
             $rq = null;
             $level = false;
+            $params = array();
             foreach ($row as $key2 => $value2) {
-                (isset($key2) && $key2 == "hc_name") ? ($hc_name = $value2 = $value2 . "_" . $i) : null;
-                $val ? $val .= ($value2 != null ? (", '" . $value2 . "'") : ", NULL") : $val .= ($value2 != null ? ("'" . $value2 . "'") : "NULL");
+                switch ($key2) {
+                    case 'hc_name':
+                        $value2 = filter_var($value2, FILTER_SANITIZE_STRING);
+                        $hc_name = $value2 = $value2 . "_" . $i;
+                        $params[] = $value2;
+                        break;
+                    case 'hc_alias':
+                    case 'hc_comment':
+                        $value2 = filter_var($value2, FILTER_SANITIZE_STRING);
+                        $params[] = $value2;
+                        break;
+                    case 'level':
+                        $value2 = filter_var($value2, FILTER_VALIDATE_INT);
+                        if ($value2) {
+                            $params[] = $value2;
+                            $level = true;
+                        } else {
+                            $params[] = null;
+                        }
+                        break;
+                    case 'icon_id':
+                        $value2 = filter_var($value2, FILTER_VALIDATE_INT);
+                        $value2
+                            ? $params[] = $value2
+                            : $params[] = null;
+                        break;
+                    case 'hc_activate':
+                        $value2 = filter_var(
+                            $value2,
+                            FILTER_VALIDATE_REGEXP,
+                            array(
+                                "options" => array(
+                                    "regexp" => "/^0|1$/"
+                                )
+                            )
+                        );
+                        $value2
+                            ? $params[] = $value2
+                            : $params[] = "0";
+                        break;
+                }
+                $val
+                    ? $val .= ", ?"
+                    : $val .= "NULL";
```

Comment:
first value, should always be null ?

Reply:
why not concatenate $val at each bound to be sure to have the right bound param

Reply:
yes it is the HostCategorie ID whose set as NULL